lib/rucio/tests/test_inequality_engine.py

```diff
@@ -0,0 +1,307 @@
+                self.assertEqual(translate(string), control)
+
+
+class TestGetNumOP(unittest.TestCase):
+
+    def test_GetNumOP(self):
+        string = "87 < test < 100"
+        self.assertEqual(get_num_op(string), 2)
+
+        string = "test == 100"
+        self.assertEqual(get_num_op(string), 1)
+
+        string = "87 <= test"
+        self.assertEqual(get_num_op(string), 1)
+
+
+class TestConvertTernary(unittest.TestCase):
+
+    def test_ConvertTernary(self):
+        string = "87 < test < 100"
+        control = ["87 < test", "test < 100"]
+        result = convert_ternary(string)
+
+        for i, r in enumerate(result):
+            self.assertEqual(r, control[i])
+
+    def test_ConvertNonTernary(self):
+        string = "87 < test"
+        control = ["87 < test"]
+        result = convert_ternary(string)
+
+        for i, r in enumerate(result):
+            self.assertEqual(r, control[i])
+
+    def test_ConvertTernary2(self):
+        string = "87 == test < 7"
+        control = ["87 == test", "test < 7"]
+        result = convert_ternary(string)
+
+        for i, r in enumerate(result):
+            self.assertEqual(r, control[i])
+
+    def test_ConvertTernary3(self):
+        string = "87 == test == 7"
+        control = ["87 == test", "test == 7"]
+        result = convert_ternary(string)
+
+        for i, r in enumerate(result):
+            self.assertEqual(r, control[i])
+
+
+class TestExpandMetadata(unittest.TestCase):
+
+    def test_ExpandMetadata(self):
+        string = "87 < length"
+        control = "87 < " + DEFAULT_MODEL + ".length"
+        self.assertEqual(control, expand_metadata(string))
+
+    def test_ExpandNoMetadata(self):
+        string = "87 < doesnt_exist"
+        self.assertEqual(string, expand_metadata(string))
+
+
+class TestConditionSplit(unittest.TestCase):
+
+    def test_ConditionSplit_A(self):
+        string = "87 < length"
+        control = ["87", "<", "length"]
+        for i, w in enumerate(condition_split(string)):
+            self.assertEqual(w, control[i])
+
+    def test_ConditionSplit_B(self):
+        string = "length >= 21"
+        control = ["length", ">=", "21"]
+        for i, w in enumerate(condition_split(string)):
+            self.assertEqual(w, control[i])
+
+
+class TestFlipIfNeeded(unittest.TestCase):
+
+    def test_FlipIfNeeded_Needed(self):
+        string = "87 < length"
+        splitted = condition_split(string)
+        splitted = flip_if_needed(splitted)
+        control = ["length", ">", "87"]
+        for i, w in enumerate(splitted):
+            self.assertEqual(w, control[i])
+
+    def test_FlipIfNeeded_NotNeeded(self):
+        string = "length < 87"
+        splitted = condition_split(string)
+        splitted = flip_if_needed(splitted)
+        control = ["length", "<", "87"]
+        for i, w in enumerate(splitted):
+            self.assertEqual(w, control[i])
+
+
+class TestInequalityEngineOffline(unittest.TestCase):
```

Comment:
o.O

Reply:
This was needed because I thought I had to use the generation of query filters using a tuple formed as `key, op, value`.
It should not be needed anymore.

Reply:
:laughing:

Reply:
🤪🤪🤪